### PR TITLE
Admit value-position object literals in the SSR seed gate (#2696 Step 1)

### PR DESCRIPTION
**Pull request by kfly8**
## Summary

Step 1 of #2696. The support gate (`checkSupport`, `expression-parser.ts`) refused `object-literal` unconditionally — correct at a RENDERED position (no template language stringifies a bare map the way it does a scalar), but the same refusal also fired at every VALUE position reachable through it, so a signal like `createSignal([{ a: 'p' }].map(t => t.a).join(','))` classified `opaque` and SSR'd empty on all 7 template-stash adapters while Hono rendered `p` (the `callback-param-shadows-prop` half of #2696; the `todo-app` half additionally needs object **spread** and stays pinned for Step 2).

## Design: position is declared at the entry point, and only there

`checkSupport` gains a `pos: 'rendered' | 'value'` parameter with **pure inheritance** — every recursive site passes `pos` through unchanged, and only the two entry points declare it: `isSupported` (rendered, public API unchanged, byte-identical refusal reason) and the new `isSupportedValue` (value). `computeSsrSeedPlan`'s `classify` uses `isSupportedValue`, since a seed is an assignment, never a render.

An earlier revision forced `'value'` onto container contents (array elements, callback receivers/bodies/args); review rejected it because that admits objects at rendered positions where they reach stringification or equality — `{[{a:1}]}`, `[{a:1}].join(',')`, `.includes({a:1})`, a rendered `.filter()`'s surviving elements — each a cross-language divergence surface (JS `[object Object]` vs Python/Perl reprs). Pure inheritance keeps rendered behavior exactly as before this PR; regression tests pin that (`isSupported` still refuses `[{a:1}]` / `[{a:1}].join(',')` / `.map(t => ({a:t}))` while `isSupportedValue` admits them).

Rebuilding on pure inheritance surfaced one hidden rendered-position assumption: each adapter's dual-purpose `convertExpressionTo<Lang>` (JSX interpolation AND derived-seed RHS) gated internally on `isSupported`, re-refusing trees the seed plan had already classified derived at value position. The same principle resolves it — the function takes a position (default `'rendered'`), and only `generateDerivedMemoSeed` passes `'value'` (each adapter's single derived-seed call site, grep-verified).

## Changes

- **`@barefootjs/jsx`**: `checkSupport(expr, pos)` + `isSupportedValue`; seed plan classifies at value position. `extractSsrDefaults`'s static evaluator gains `.map()` (single-param expression-bodied arrow over a resolved array; anything else stays UNRESOLVED) and property reads on resolved plain-object bases — needed because a constant-frees derived step emits no in-template recompute, so the static default is the value production reads.
- **7× template-stash adapters**: `objectLiteral` `ParsedExprEmitter` lowers populated literals to native dict/hash/hashref syntax (previously only the empty `?? {}` fallback was reachable); `convertExpressionTo<Lang>` threads `pos`.
- **`@barefootjs/go-template`**: no map-literal template syntax exists, so object literals lower through a new variadic `bf_map` runtime helper (object counterpart of `bf_arr`), reusing `eval.go`'s existing `Env` builder.
- **Fixtures** (change-time coupling): `callback-param-shadows-prop` graduates off `renderDivergences` on all 7 adapters (its fixture flips into the regression test); new `map-object-literal-body` exercises an object-literal `.map()` body with a non-empty free set — a genuine in-template recompute on every backend, and the case that caught the emitter-gate regression during review.
- Locks regenerated after the CI-identical full dist build; changeset extended (all 8 packages).

## Test results

All serial at HEAD: jsx 3098 pass (2 pre-existing order-dependent fails, baseline-identical); blade 1478/0, jinja 1478/0, twig 1486/0, rust 1476/0, go-template 1786/0, hono 1559/0; erb 1439 pass / mojolicious 1637 pass / xslate 1477 pass with only the pre-existing `date-tolocale-named-tz` environment failures (missing tz data locally); adapter-tests meta 2122/0; compat 502/0 with `git diff -- ui/` empty after regeneration (lock freshness verified against fresh dist builds).

`todo-app` / `todo-app-ssr` stay pinned — object spread is #2696 Step 2.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_0198yG2px5ygfvmdoMkhRS13

---
_Generated by [Claude Code](https://claude.ai/code/session_0198yG2px5ygfvmdoMkhRS13)_

### CODE REVIEW

File: packages/adapter-mojolicious/src/adapter/expr/emitters.ts [hunk 1]
```diff
@@ -700,16 +700,14 @@ export class MojoTopLevelEmitter implements ParsedExprEmitter {
...
+  objectLiteral(properties: ObjectLiteralProperty[], _raw: string, emit: (e: ParsedExpr) => string): string {
+    // Reachable here in VALUE position (a `.map()` receiver/callback body, an
+    // array-literal element, …) since `isSupportedValue` admits an object
+    // literal whose every property value is itself supported
+    // (expression-parser.ts, `checkSupport`'s `pos` parameter). Emit a Perl
+    // hashref literal, quoted the same way `objectLiteralToPerlHashref`
+    // quotes the spread-path literal.
+    if (properties.length === 0) return '{}'
+    return `{ ${properties.map(p => `'${p.key.replace(/'/g, "\\'")}' => ${emit(p.value)}`).join(', ')} }`
   }
```
[the remark this answers is not in the document]
Comment (kfly8) [reply packages/adapter-mojolicious/src/adapter/expr/emitters.ts:713]: Fixed in 918adf3b2 (backslash-first, then quote), centralized as `escapePerlSingleQuoted` in `perl-naming.ts` and applied to `perlHashKey` (same pre-existing bug), the `objectLiteral` emitter, and `spread-codegen.ts`'s two inline sites. Horizontal sweep across the other adapters' key escapers: blade/jinja/minijinja/twig/xslate were already backslash-first correct; ERB's `rubySymbolKey` double-quote fallback had the identical bug and got the same fix (`escapeRubyDoubleQuoted`). Unit tests cover `\` + quote keys on both.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/jsx/src/ssr-defaults.ts [hunk 2]
```diff
@@ -647,12 +647,22 @@ function evalNode(node: ts.Expression, ctx: EvalContext): EvalResult {
...
+    const baseResult = evalNode(node.expression, ctx)
     // `props.X` / `props?.X` — read of a binding we know nothing about, so
     // resolve to `undefined`. Chained access (`a.b.c`) collapses the same way
     // because the base read is already undefined.
-    const baseResult = evalNode(node.expression, ctx)
     if (baseResult === undefined) return undefined
-    return UNRESOLVED
+    if (baseResult === UNRESOLVED || baseResult === null || typeof baseResult !== 'object') {
+      return UNRESOLVED
+    }
```
[the remark this answers is not in the document]
Comment (kfly8) [reply packages/jsx/src/ssr-defaults.ts:657]: Fixed in 918adf3b2: the guard now requires the base's prototype to be `Object.prototype`/`null`; any other base (arrays included) returns UNRESOLVED. `isElementAccessExpression` (numeric indexing on arrays, JS-faithful) is unchanged. Tests pin both directions: an array-element property access aborts the computation to UNRESOLVED, a plain-object missing key still resolves to `undefined`.

---
_Generated by [Claude Code](https://claude.ai/code)_
